osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -113,21 +114,25 @@ public virtual string SettingDescription
         [JsonIgnore]
         public virtual Type[] IncompatibleMods => Array.Empty<Type>();
 
-        private IReadOnlyList<IBindable>? settingsBacking;
+        private IReadOnlyDictionary<string, IBindable>? settingsBacking;
 
         /// <summary>
-        /// A list of the all <see cref="IBindable"/> settings within this mod.
+        /// All <see cref="IBindable"/> settings within this mod.
         /// </summary>
-        internal IReadOnlyList<IBindable> Settings =>
+        internal IEnumerable<IBindable> SettingsBindables => Settings.Values;
```

Comment:
The order of `Dictionary<TKey,TValue>.Values` [is unspecified](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.keys?view=net-7.0#remarks). If it was only iterated over, then maybe that wouldn't matter, but it's also being used for equality comparisons. Seems dangerous to me.

Should probably be manually sorted by key.

Reply:
For the equals check it shouldn't matter that it isn't sorted. If the types are the same then the order will also be the same no matter what.
`return GetType() == other.GetType() && SettingsBindables.SequenceEqual(other.SettingsBindables, ModSettingsEqualityComparer.Default);`

Reply:
> If the types are the same then the order will also be the same no matter what.

That relies on implementation details of `Dictionary`, does it not? It may be currently *true*, because `ValueCollection` may be implemented in a particular way that always makes it so (e.g. iterates over hash buckets or whatever), but it doesn't make it any less dangerous as the behaviour shouldn't be relied upon.

Reply:
> That relies on implementation details of Dictionary, does it not?

Yes it does, in that case something like checking their lengths and if items from one dictionary exist in the other should suffice.

Reply:
Implementation suggested above is buggy as it does not consider cases where `other.Settings` has keys `this.Settings` does not.

Sorting is much simpler and solves the issue at hand correctly.